### PR TITLE
[zh-cn] Update 'sending forms through javascript'

### LINKED ISSUE #16191: 拼写错误：第一条备注里面的「有点」应改为「优点」
### MDN URL

https://developer.mozilla.org/zh-CN/docs/Learn/Forms/Sending_forms_through_JavaScript

### 与此议题相关的小节或标题？

_No response_

### 哪些信息是错误、无用或不完善的？

备注： 如今 [Fetch API](https://developer.mozilla.org/zh-CN/docs/Web/API/Fetch_API) 常用于取代 XHR——它是 XHR 的更现代、更新的版本，其工作原理类似，但具有一些有点。你将在本文中看到的大多数 XHR 代码都可以替换为 Fetch。

### 你期望看到什么？

备注： 如今 [Fetch API](https://developer.mozilla.org/zh-CN/docs/Web/API/Fetch_API) 常用于取代 XHR——它是 XHR 的更现代、更新的版本，其工作原理类似，但具有一些优点。你将在本文中看到的大多数 XHR 代码都可以替换为 Fetch。

### 你有什么支持的链接、参考或引用？

Note: The [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) is often used in place of XHR these days — it is a modern, updated version of XHR, which works similarly but has some advantages. Most of the XHR code you'll see in this article could be swapped out for Fetch.

英文原文对应的单词是 advantages(优点)

### 你还想补充什么内容？

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `zh-cn/learn/forms/sending_forms_through_javascript`
* MDN URL: https://developer.mozilla.org/zh-CN/docs/Learn/Forms/Sending_forms_through_JavaScript
* GitHub URL: https://github.com/mdn/translated-content/blob/main/files/zh-cn/learn/forms/sending_forms_through_javascript/index.md
* Last commit: https://github.com/mdn/translated-content/commit/3f4a97f9f50d4acf0c84558ac9d02469d3589e2c
* Document last modified: 2023-08-31T02:30:56.000Z

</details>

### PR DESCRIPTION
**Pull request by jasonren0403**
### Description

update document and fix errors

### Related issues and pull requests

closes #16191

### CODE REVIEW

File: files/zh-cn/learn/forms/sending_forms_through_javascript/index.md [hunk 1]
```diff
@@ -86,47 +95,53 @@ function sendData(data) {
...
 {{EmbedLiveSample("构建_xmlhttprequest", "100%", 50)}}
 
-> **备注：** 当你想要往第三方网站传输数据时，使用{{domxref("XMLHttpRequest")}}会受到同源策略的影响。如果你需要执行跨域请求，你需要熟悉一下[CORS 和 HTTP 访问控制](/zh-CN/docs/HTTP/Access_control_CORS).
+> **备注：** 当你想要往第三方网站传输数据时，使用 {{domxref("XMLHttpRequest")}} 会受到{{glossary('same-origin policy', '同源策略')}}的影响。如果你需要执行跨域请求，你需要熟悉一下 [CORS 和 HTTP 访问控制](/zh-CN/docs/Web/HTTP/CORS)。
 
-### 使用 XMLHttpRequest 和 the FormData object（表单数据对象）
+### 使用 XMLHttpRequest 和 FormData 对象
 
-手动建立一个 HTTP 请求非常困难。幸运的是，最近的 [XMLHttpRequest 规范](http://www.w3.org/TR/XMLHttpRequest/)提供了一种方便简单的方法 — 利用{{domxref("XMLHttpRequest/FormData","FormData")}}对象来处理表单数据请求。
+手动建立一个 HTTP 请求非常困难。幸运的是，[XMLHttpRequest 规范](http://www.w3.org/TR/XMLHttpRequest/)提供了一种方便简单的方法——利用 {{domxref("FormData","FormData")}} 对象来处理表单数据请求。
```
Comment (yin1999) [inline files/zh-cn/learn/forms/sending_forms_through_javascript/index.md:112]:
```suggestion
手动建立一个 HTTP 请求非常困难。幸运的是，[XMLHttpRequest 规范](https://www.w3.org/TR/XMLHttpRequest/)提供了一种方便简单的方法——利用 {{domxref("FormData","FormData")}} 对象来处理表单数据请求。
```

File: files/zh-cn/learn/forms/sending_forms_through_javascript/index.md [hunk 2]
```diff
@@ -394,35 +348,53 @@ window.addEventListener("load", function () {
...
+- [如何构建 HTML 表单](/zh-CN/docs/Learn/Forms/How_to_structure_a_web_form)
+- [原生表单控件](/zh-CN/docs/Learn/Forms/Basic_native_form_controls)
+- [HTML5 输入（input）类型](/zh-CN/docs/Learn/Forms/HTML5_input_types)
+- [其他表单控件](/zh-CN/docs/Learn/Forms/Other_form_controls)
+- [UI 伪类](/zh-CN/docs/Learn/Forms/UI_pseudo-classes)
+- [样式化 HTML 表单](/zh-CN/docs/Learn/Forms/Styling_web_forms)
+- [表单数据校验](/zh-CN/docs/Learn/Forms/Form_validation)
+- [发送表单数据](/zh-CN/docs/Learn/Forms/Sending_and_retrieving_form_data)
+
+### 更进一步
```
[the remark this answers is not in the document]
Comment (jasonren0403) [reply files/zh-cn/learn/forms/sending_forms_through_javascript/index.md:394]: maybe "进阶主题" is better here(like what we used in sidebar 🤔 ), further PR is needed for current files to unify the wording

File: files/zh-cn/learn/forms/sending_forms_through_javascript/index.md [hunk 3]
```diff
@@ -394,35 +348,53 @@ window.addEventListener("load", function () {
...
+这里是实时演示效果：
 
 {{EmbedLiveSample("处理二进制数据", "100%", 150)}}
 
 ## 总结
 
-取决于不同的浏览器，通过 JavaScript 发送数据可能会很简单，也可能会很困难。{{domxref("XMLHttpRequest/FormData","FormData")}} 对象是通用的答案，所以请毫不犹豫的在旧浏览器上通过 polyfill 使用它：
+取决于不同的浏览器和正在处理数据的类型，通过 JavaScript 发送数据可能会很简单，也可能会很困难。{{domxref("FormData","FormData")}} 对象是通用的答案，所以请毫不犹豫地在旧浏览器上通过 [polyfill](https://github.com/jimmywarting/FormData) 使用它：
+
+## 参见
```
Comment (jasonren0403) [inline files/zh-cn/learn/forms/sending_forms_through_javascript/index.md:380]: BTW, I don't think we really need 'see also' part like this... we've got the sidebars, and the content are just identical 🤔 

image

File: files/zh-cn/learn/forms/sending_forms_through_javascript/index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (moshuying) [reply files/zh-cn/learn/forms/sending_forms_through_javascript/index.md:394]: "进阶主题" more like develope docs

**Review by moshuying — APPROVED** [review]

File: files/zh-cn/learn/forms/sending_forms_through_javascript/index.md (hunk 3 above)
Comment (yin1999) [reply files/zh-cn/learn/forms/sending_forms_through_javascript/index.md:380]: Similar to #11653. Could you open a discussion for this ;)

Comment (jasonren0403) [reply files/zh-cn/learn/forms/sending_forms_through_javascript/index.md:380]: Discussion opened.

Comment (jasonren0403) [reply files/zh-cn/learn/forms/sending_forms_through_javascript/index.md:380]: I'm going to resolve this because this is out of the PR's scope 👀

**Review by yin1999 — APPROVED** [review]
🙏 @jasonren0403

**Pull request merged**